### PR TITLE
Add GenericTemplateView helper

### PR DESCRIPTION
**Pull request by lenarother**

### CODE REVIEW

File: template_helpers/views.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import os
+
+from django.views.generic import TemplateView
+
+
+class GenericTemplateView(TemplateView):
+
+    def get_template_names(self):
+        template_filename = '%s.html' % (self.kwargs['template'] or 'index')
+        return [os.path.join('tests', template_filename)]
```
Comment (stephrdev) [inline template_helpers/views.py]: The base directory ("tests" should be changeable via as_view kwarg. We might also add a "get_base_directory()" method to allow further overriding the default behaviour.

**Review by stephrdev — COMMENTED** [review]
We should add some docs on how to use the GenericTemplateView and what it could be used for. Thanks for starting to backport this feature!

**Comment by lenarother** [comment]
Hi @stephrdev ,
I added get_base_directory method and documentation.

LG,
Magda

File: template_helpers/views.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import os
+
+from django.views.generic import TemplateView
+
+
+class GenericTemplateView(TemplateView):
+
```
Comment (stephrdev) [inline template_helpers/views.py:8]: base_template_dir = ''

Comment (lenarother) [reply template_helpers/views.py:8]: Done

File: docs/usage.rst [hunk 3]
```diff
@@ -64,22 +64,26 @@ The output will be
...
             ),
 
-If test templates are located in templates/tests/...
-(e.g. templates/tests/base/buttons/buttons.html) we can now hit them by calling
+If test templates are located in templates/mytests/...
+(e.g. templates/mytests/base/buttons/buttons.html) we can now hit them by calling
 e.g. localhost:8000/tests/base/buttons/buttons url.
 
-If no base_dir or template names are given, the view will try to render index.html.
-For more elaborate behavior the ``get_template_names`` method can be overwritten.
```
Comment (stephrdev) [inline docs/usage.rst]: Both, the get_template_names and get_template_base_dir method can be overwritten.

File: template_helpers/views.py [hunk 4]
```diff
@@ -4,9 +4,10 @@
 
 
 class GenericTemplateView(TemplateView):
+    base_template_dir = ''
 
     def get_base_directory(self, **kwargs):
```
Comment (stephrdev) [inline template_helpers/views.py]: Rename to get_template_base_dir.

File: template_helpers/views.py [hunk 5]
```diff
@@ -4,9 +4,10 @@
...
-        return self.kwargs.get('base_dir') or ''
+        return self.kwargs.get('base_template_dir') or self.base_template_dir
```
Comment (stephrdev) [inline template_helpers/views.py]: I think I don't want to support changing the base directory via url parameters. Lets rely on the as_view option/property.

Comment (lenarother) [reply template_helpers/views.py]: Done.

File: template_helpers/views.py (hunk 4 above)
Comment (lenarother) [reply template_helpers/views.py]: Done. We now have get_template_base_dir and template_base_dir for consistency.

File: docs/usage.rst (hunk 3 above)
Comment (lenarother) [reply docs/usage.rst]: Done.

File: template_helpers/views.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import os
+
+from django.views.generic import TemplateView
+
+
+class GenericTemplateView(TemplateView):
+    template_base_dir = ''
+
+    def get_template_base_dir(self, **kwargs):
```
Comment (stephrdev) [inline template_helpers/views.py]:
```suggestion
    def get_template_base_dir(self):
```

**Review by stephrdev — COMMENTED** [review]
Please update the method signature. After that, ready for merge :)

**Pull request merged**